spark/src/main/scala/org/apache/comet/expressions/CometCast.scala

```diff
@@ -70,9 +70,13 @@ object CometCast {
           case _ =>
             Unsupported
         }
-      case (_: DecimalType, _: DecimalType) =>
-        // https://github.com/apache/datafusion-comet/issues/375
-        Incompatible()
+      case (from: DecimalType, to: DecimalType) =>
+        if (to.precision < from.precision) {
+          // https://github.com/apache/datafusion/issues/13492
+          Incompatible()
```

Comment:
We can add extra info here to show to the user.

```suggestion
          Incompatible(Some("Casting to smaller precision is not supported"))
```